tools/xtce_generation/telemetry_generator.py

```diff
@@ -135,7 +142,11 @@ def get_unique_bits_length(self):
         unique_lengths = dict(sorted(unique_lengths.items(), key=lambda item: item[1]))
         return unique_lengths
 
-    def create_parameter_types(self, parameter_type_set, unique_lengths):
+    def create_parameter_types(
+        self,
+        parameter_type_set: list,
+        unique_lengths: dict,
+    ) -> list:
```

Comment:
I did some digging here and ran this code with a `print(type(parameter_type_set))` to see what type this was. I think this will solve the issues.

```suggestion
    def create_parameter_types(
        self,
        parameter_type_set: Et.Element,
        unique_lengths: dict,
    ) -> Et.Element:
```

Reply:
And in the docstrings (line 158 and 160) you can change the type from `list TODO Double Check` to `Et.Element` for `parameter_type_set` and to `dict` for `unique_lengths`